packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -64,71 +121,197 @@ export class AtomicCommerceProductList
+      this.density,
+      this.imageSize,
+
+      this.productState?.isLoading,
+      displayPlaceholders
+    );
   }
 
-  private generateStarRating(rating: number) {
-    const stars = [];
-    for (let i = 0; i < 5; i++) {
-      stars.push(
-        <atomic-icon
-          class="w-5"
-          style={
-            i < rating ? {fill: 'gray'} : {fill: 'white', stroke: 'lightgray'}
+  private getPropsForAtomicProduct(product: Product) {
+    return {
+      // TODO: add back once interactive result is implemented for products
+      /* interactiveResult: buildInteractiveResult(this.bindings.engine, {
+        options: {result},
+      }), */
+      product,
+      renderingFunction: this.itemRenderingFunction,
+      loadingFlag: this.loadingFlag,
+      key: this.productListCommon.getResultId(
+        product.permanentid,
+        this.productState.responseId,
+        this.density,
+        this.imageSize
+      ),
+      content: this.productTemplateProvider.getTemplateContent(product),
+      store: this.bindings.store,
+      density: this.density,
+      imageSize: this.imageSize,
+      display: this.display,
+    };
+  }
+
+  private renderAsGrid() {
+    return this.productState.products.map((product, i) => {
+      const propsForAtomicProduct = this.getPropsForAtomicProduct(product);
+      return (
+        <DisplayGrid
+          item={{
+            ...product,
+            clickUri: product.clickUri,
+            title: product.ec_name ?? 'temp',
+          }}
+          // TODO: add back once the interactive result is implemented
+          //{...propsForAtomicProduct.interactiveResult}
+          // TODO: Remove these back once the interactive result is implemented
+          setRef={(element) =>
+            element && this.productListCommon.setNewResultRef(element, i)
           }
-          icon={StarIcon}
-        ></atomic-icon>
+          select={function (): void {
+            throw new Error('Function not implemented.');
+          }}
+          beginDelayedSelect={function (): void {
+            throw new Error('Function not implemented.');
+          }}
+          cancelPendingSelect={function (): void {
+            throw new Error('Function not implemented.');
+          }}
+        >
+          <atomic-product {...this} {...propsForAtomicProduct}></atomic-product>
+        </DisplayGrid>
       );
-    }
-
-    return stars;
+    });
   }
 
-  // TODO: Refactor to support result templates
-  // TODO: Refactor to use guards/wrappers as in atomic-result-list
-  public render() {
-    const products =
-      this.bindings.interfaceElement.type === 'product-listing'
-        ? this.productListingState.products
-        : this.searchState.products;
-
-    if (
-      products.length === 0 &&
-      !this.productListingState?.isLoading &&
-      !this.searchState?.isLoading
-    ) {
-      return <div>No products found.</div>;
+  private renderAsTable() {
+    if (this.productState.products.length > 0) {
```

Comment:
Hum wouldn't it be the opposite ? `<= 0`  🤔

Reply:
Actually removed that part :)